etc/spec/SPECIFICATION.md

```diff
@@ -117,12 +114,25 @@ is verified as follows:
  - The symbols . , / ? ~ ! @ # $ % ^ & * ( ) - _ = + [ ] { } \ | ' " ` ; :
 3. The username must not be reserved.
 
-The username is also validated for form, but not for reservation, as the
-username is being typed.
+A username is reserved if and only if someone is registered with that username
+or someone use it at that moment(even if that username is not registered).
+
+While the user is typing and the username is valid we have two cases.
+Either the username is registered or it is not.
+
+1. If the username is registered then a new form is appeared which it asks from the
+   user to enter the password that is related with the username. If the password
```

Comment:
Not sure how you plan to change the form **while the user is typing**. You should explain this in the spec.